etc/eslint/rules/typescript.js

```diff
@@ -0,0 +1,2526 @@
+* @name no-unnecessary-qualifier
+* @memberof rules
+* @type {string}
+* @default 'error'
+* @see [no-unnecessary-qualifier]{@link https://typescript-eslint.io/rules/no-unnecessary-qualifier}
+*
+* @example
+* // Bad...
+* namespace A {
+*   export const B = 1;
+* }
+* const x = A.B;
+*
+* // Good...
+* const B = 1;
+* const x = B;
+*/
+rules[ '@typescript-eslint/no-unnecessary-qualifier' ] = 'error';
+
+/**
+* Disallows unnecessary type arguments in generics where the argument is the default.
+*
+* @name no-unnecessary-type-arguments
+* @memberof rules
+* @type {string}
+* @default 'error'
+* @see [no-unnecessary-type-arguments]{@link https://typescript-eslint.io/rules/no-unnecessary-type-arguments}
+*
+* @example
+* // Bad...
+* const x: Array<number> = [];
+*
+* // Good...
+* const x: number[] = [];
+*/
+rules[ '@typescript-eslint/no-unnecessary-type-arguments' ] = 'error';
+
+/**
+* Disallows unnecessary type assertions that do not change the type of an expression.
+*
+* @name no-unnecessary-type-assertion
+* @memberof rules
+* @type {string}
+* @default 'error'
+* @see [no-unnecessary-type-assertion]{@link https://typescript-eslint.io/rules/no-unnecessary-type-assertion}
+*
+* @example
+* // Bad...
+* const x: number = 10;
+* const y = x as number;
+*
+* // Good...
+* const x: number = 10;
+* const y = x;
+*/
+rules[ '@typescript-eslint/no-unnecessary-type-assertion' ] = 'error';
+
+/**
+* Disallows unused expressions which have no effect on the state of the program.
+*
+* @name no-unused-expressions
+* @memberof rules
+* @type {string}
+* @default 'error'
+* @see [no-unused-expressions]{@link https://typescript-eslint.io/rules/no-unused-expressions}
+*
+* @example
+* // Bad...
+* 0;
+* if(false) {}
+*
+* // Good...
+* (() => {
+*   const a = 0;
+* })();
+*/
+rules[ '@typescript-eslint/no-unused-expressions' ] = 'error';
+
+/**
+* Disables the rule against unused variables.
+*
+* @name no-unused-vars
+* @memberof rules
+* @type {string}
+* @default 'off'
+* @see [no-unused-vars]{@link https://typescript-eslint.io/rules/no-unused-vars}
+*
+* @example
+* // Okay...
+* const unused = "I'm never used";
+*/
+rules[ '@typescript-eslint/no-unused-vars' ] = 'off';
+
+/**
+* Disallows the use of variables before they are defined.
+*
+* @name no-use-before-define
+* @memberof rules
+* @type {string}
+* @default 'error'
+* @see [no-use-before-define]{@link https://typescript-eslint.io/rules/no-use-before-define}
+*
+* @example
+* // Bad...
+* console.log(a);
+* let a = 10;
+*
+* // Good...
+* let b = 10;
+* console.log(b);
+*/
+rules[ '@typescript-eslint/no-use-before-define' ] = 'error';
+
+/**
+* Disallows the use of `require` statements except in import statements.
+*
+* @name no-var-requires
+* @memberof rules
+* @type {string}
+* @default 'error'
+* @see [no-var-requires]{@link https://typescript-eslint.io/rules/no-var-requires}
+*
+* @example
+* // Bad...
+* const moduleA = require('moduleA');
+*
+* // Good...
+* import moduleA from 'moduleA';
```

Comment:
We should add an example here that is allowed:

```
import moduleA = require( 'moduleA' );
```

Reply:
Done.